# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1383852905 18000
#      Thu Nov 07 14:35:05 2013 -0500
# Node ID 8b7bbe021d0fb8f8f74a3c2346ec8066027dda07
# Parent  0d660785098ef172aded9f8d63057fc703cd7290
refuse to attempt to fix stl_algo.h except for g++ 4.8.2

* acinclude.m4 (OCTAVE_CHECK_BROKEN_STL_ALGO_H): Don't set cache
variable to yes unless it appears that we are using g++ 4.8.2.

diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -2113,30 +2113,32 @@ return v[3] == 207089 ? 0 : 1;
     AC_LANG_POP(C++)
   ])
   if test "$GXX" = yes; then
     if test $octave_cv_broken_stl_algo_h = yes; then
       case "$GXX_VERSION" in
         4.8.2)
         ;;
         *)
-          warn_stl_algo_h="UNEXPECTED: found stl_algo.h broken in $GXX_VERSION"
+          octave_cv_broken_stl_algo_h=no
+          warn_stl_algo_h="UNEXPECTED: found nth_element broken in $GXX_VERSION.  Refusing to fix except for g++ 4.8.2."
           OCTAVE_CONFIGURE_WARNING([warn_stl_algo_h])
         ;;
       esac
     else
       case "$GXX_VERSION" in
         4.8.2)
-          warn_stl_algo_h="UNEXPECTED: found stl_algo.h working in g++ $GXX_VERSION -- has it been patched on your system?"
+          warn_stl_algo_h="UNEXPECTED: found nth_element working in g++ 4.8.2.  Has it been patched on your system?"
           OCTAVE_CONFIGURE_WARNING([warn_stl_algo_h])
         ;;
       esac
     fi
   else
-    warn_stl_algo_h="UNEXPECTED: nth_element test failed (expected only for g++ 4.8.2)"
+    octave_cv_broken_stl_algo_h=no
+    warn_stl_algo_h="UNEXPECTED: nth_element test failed.  Refusing to fix except for g++ 4.8.2."
     OCTAVE_CONFIGURE_WARNING([warn_stl_algo_h])
   fi
 ])
 
 dnl         End of macros written by Octave developers
 dnl ------------------------------------------------------------
 dnl
 
